src/main/java/org/apache/commons/lang3/text/ExtendedMessageFormat.java

```diff
@@ -483,15 +514,16 @@ private StringBuilder appendQuotedString(final String pattern, final ParsePositi
         }
         next(pos);
 
-        final int start = pos.getIndex();
+        final @LTEqLengthOf("pattern") int start = pos.getIndex(); // #1
         final char[] c = pattern.toCharArray();
-        final int lastHold = start;
-        for (int i = pos.getIndex(); i < pattern.length(); i++) {
-            if (c[pos.getIndex()] == QUOTE) {
+        final @LTEqLengthOf("c") int lastHold = start;
+        for (@IndexOrHigh("c") int i = pos.getIndex(); i < pattern.length(); i++) { // #2
+            switch (c[pos.getIndex()]) { // #3
```

Comment:
Why change this `if` to a `switch`? I think they're semantically equivalent, but it clutters the diff.